src/cross_section_measurement/04_make_plots_matplotlib.py

```diff
@@ -615,6 +623,79 @@ def make_plots( histograms, category, output_folder, histname, show_ratio = True
+        x_limits = axes.get_xlim()
+        ax2.set_xlim(x_limits)
+
+        ax2.yaxis.set_major_locator( MultipleLocator( 0.5 ) )
+        if not variable in ['NJets']:
+            ax2.yaxis.set_minor_locator( MultipleLocator( 0.1 ) )
+
+        if variable in ['NJets', 'abs_lepton_eta', 'lepton_eta']:
+            plt.xlabel('$%s$' % variables_latex[variable], CMS.x_axis_title )
+        else:
+            plt.xlabel( '$%s$ [GeV]' % variables_latex[variable], CMS.x_axis_title )
+
+        plt.tick_params( **CMS.axis_label_major )
+        if not variable in ['NJets']:
+            plt.tick_params( **CMS.axis_label_minor )
+        plt.ylabel( '$\\frac{\\textrm{generator}}{\\textrm{central}}$', CMS.y_axis_title )
+        ax2.yaxis.set_label_coords(-0.115, 0.8)
+        #draw a horizontal line at y=1 for data
+        plt.axhline(y = 1, color = 'black', linewidth = 2)
+
+        central_mc = histograms['powhegPythia8']
+        for key, hist in sorted( histograms.iteritems() ):
+            if not 'unfolded' in key and not 'measured' in key:
+                ratio = hist.Clone()
+                ratio.Divide( central_mc ) #divide by data
+                line, h = rplt.hist( ratio, axes = ax2, label = 'do_not_show' )
+                if dashes[key] != None:
+                    h.set_dashes(dashes[key])
+
+        if variable == 'MET':
+            ax2.set_ylim( ymin = 0.8, ymax = 1.2 )
+            ax2.yaxis.set_major_locator( MultipleLocator( 0.5 ) )
+#             ax2.yaxis.set_minor_locator( MultipleLocator( 0.1 ) )
+        if variable == 'MT':
+            ax2.set_ylim( ymin = 0.8, ymax = 1.2 )
+            ax2.yaxis.set_major_locator( MultipleLocator( 0.2 ) )
+            ax2.yaxis.set_minor_locator( MultipleLocator( 0.1 ) )
+        elif variable == 'HT':
+            ax2.set_ylim( ymin = 0.8, ymax = 1.37 )
+            ax2.yaxis.set_major_locator( MultipleLocator( 0.2 ) )
+            ax2.yaxis.set_minor_locator( MultipleLocator( 0.1 ) )
+        elif variable == 'ST':
+            ax2.set_ylim( ymin = 0.7, ymax = 1.5 )
+            ax2.yaxis.set_major_locator( MultipleLocator( 0.5 ) )
+            ax2.yaxis.set_minor_locator( MultipleLocator( 0.1 ) )
+        elif variable == 'WPT':
+            ax2.set_ylim( ymin = 0.8, ymax = 1.2 )
+            ax2.yaxis.set_major_locator( MultipleLocator( 0.5 ) )
+            ax2.yaxis.set_minor_locator( MultipleLocator( 0.1 ) )
+        elif variable == 'NJets':
+            ax2.set_ylim( ymin = 0.7, ymax = 1.5 )
+        elif variable == 'abs_lepton_eta':
+            ax2.set_ylim( ymin = 0.8, ymax = 1.2 )
+            ax2.yaxis.set_major_locator( MultipleLocator( 0.2 ) )
+            ax2.yaxis.set_minor_locator( MultipleLocator( 0.1 ) )
+        elif variable == 'lepton_pt':
+            ax2.set_ylim( ymin = 0.8, ymax = 1.3 )
+            ax2.yaxis.set_major_locator( MultipleLocator( 0.2 ) )
+            ax2.yaxis.set_minor_locator( MultipleLocator( 0.1 ) )
+
+
+
```

Comment:
This is getting quite long, so might be worth outsourcing to tools/<somefile>.py